src/snooze/ecs/minigame/vermins/VerminsMiniGameSystem.cpp

```diff
@@ -0,0 +1,126 @@
+#include <snooze/Precomp.h>
+#include <snooze/ecs/minigame/vermins/VerminsMiniGameSystem.h>
+
+#include <iostream>
+#include <random>
+
+#include <forge/engine/data/api/DataAPI.h>
+#include <forge/engine/ecs/Entity.h>
+
+#include <snooze/data/DataList.h>
+#include <snooze/data/EntityCatalog.h>
+
+//----------------------------------------------------------------------------
+VerminsMiniGameSystem::VerminsMiniGameSystem()
+    : m_ToolAcquired(false)
+    , m_Tool(nullptr)
+    , m_MaxVermins(0)
+    , m_TimeToVermin(5000)
+    , m_FirstRun (true)
+{
+}
+
+//----------------------------------------------------------------------------
+void VerminsMiniGameSystem::Execute(const u64& _dt, const forge::Entity::Ptr& _entity)
+{
+    static std::random_device rd;
+    VerminsMiniGameComponent& comp = _entity->GetComponent<VerminsMiniGameComponent>();
+
+    BaseMiniGame::Update(comp);
+
+    // Spawn new vermin, restart Timer
+    if (m_FirstRun or (m_Timer.IsElapsed() && m_Vermins.size() <= m_MaxVermins))
+    {
+        m_FirstRun = false;
+        forge::Entity::Ptr vermin = forge::DataAPI::GetDataFrom<EntityCatalog>(DataList::Entity::VerminEnemy);
+        vermin->SetPosition(rd() % (57 - 40 + 1) + 40, rd() % (57 - 40 + 1) + 40, 10.f);
+        RequestAddEntity(vermin);
+        m_Vermins.push_back({vermin, false});
+        m_Timer.Start(m_TimeToVermin);
+    }
+
+    // Lock'n'loaded?
+    if (m_ToolAcquired)
+    {
+        if (m_Tool != nullptr)
+        {
+            RequestRemoveEntity(m_Tool);
```

Comment:
In this case it works to immediately remove the tool, but if we add the feature to show an inventory slot with the currently equiped item, it will instantly be removed here.
It would be more future proof to remove if only when the minigame is over I think.

Reply:
Yes, but we need some kind of visual information to inform the player that he's ready to smack some mouses.... Maybe displaying another sprite?

Reply:
Roulyo just finished an inventory slot system, I think we can add it there

Reply:
Yes definitely. I'll rebase master and add it to the inventory system.